### ISSUE TITLE
spotcheck white error

### ISSUE THREAD

**Issue by kyungjoo-kim** [kokkos/kokkos-kernels#285]
Is this known issue ? The compiler confuses std::swap with Kokkos::swap.

```
/home/projects/ppc64le/gcc/6.4.0/include/c++/6.4.0/bits/stl_algobase.h(148): error: more than one instance of overloaded function "std::swap" matches the argument list:                                                
            function template "void Kokkos::swap(T &, T &)"                                                                                                                                                             
            function template "std::enable_if<std::__and_<std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp>>::value, void>::type std::swap(_Tp &, _Tp &)"                                                   
            argument types are: (KokkosKernels::Impl::Edge<int, Kokkos::complex<double>>, KokkosKernels::Impl::Edge<int, Kokkos::complex<double>>)                                                                      
          detected during:                                                                                                                                                                                              
            instantiation of "void std::iter_swap(_ForwardIterator1, _ForwardIterator2) [with _ForwardIterator1=__gnu_cxx::__normal_iterator<KokkosKernels::Impl::Edge<int, Kokkos::complex<double>> *, std::vector<Kok\
kosKernels::Impl::Edge<int, Kokkos::complex<double>>, std::allocator<KokkosKernels::Impl::Edge<int, Kokkos::complex<double>>>>>, _ForwardIterator2=__gnu_cxx::__normal_iterator<KokkosKernels::Impl::Edge<int, Kokkos::\
complex<double>> *, std::vector<KokkosKernels::Impl::Edge<int, Kokkos::complex<double>>, std::allocator<KokkosKernels::Impl::Edge<int, Kokkos::complex<double>>>>>]"                                                    
/home/projects/ppc64le/gcc/6.4.0/include/c++/6.4.0/bits/stl_algo.h(84): here                                                                                                                                            
            instantiation of "void std::__move_median_to_first(_Iterator, _Iterator, _Iterator, _Iterator, _Compare) [with _Iterator=__gnu_cxx::__normal_iterator<KokkosKernels::Impl::Edge<int, Kokkos::complex<double\
>> *, std::vector<KokkosKernels::Impl::Edge<int, Kokkos::complex<double>>, std::allocator<KokkosKernels::Impl::Edge<int, Kokkos::complex<double>>>>>, _Compare=__gnu_cxx::__ops::_Iter_less_iter]"                      
/home/projects/ppc64le/gcc/6.4.0/include/c++/6.4.0/bits/stl_algo.h(1919): here                                                                                                                                          
            instantiation of "_RandomAccessIterator std::__unguarded_partition_pivot(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator=__gnu_cxx::__normal_iterator<KokkosKernels::Im\
pl::Edge<int, Kokkos::complex<double>> *, std::vector<KokkosKernels::Impl::Edge<int, Kokkos::complex<double>>, std::allocator<KokkosKernels::Impl::Edge<int, Kokkos::complex<double>>>>>, _Compare=__gnu_cxx::__ops::_I\
ter_less_iter]"                                                                                                                                                                                                         
/home/projects/ppc64le/gcc/6.4.0/include/c++/6.4.0/bits/stl_algo.h(1950): here                                                                                                                                          
            instantiation of "void std::__introsort_loop(_RandomAccessIterator, _RandomAccessIterator, _Size, _Compare) [with _RandomAccessIterator=__gnu_cxx::__normal_iterator<KokkosKernels::Impl::Edge<int, Kokkos:\
:complex<double>> *, std::vector<KokkosKernels::Impl::Edge<int, Kokkos::complex<double>>, std::allocator<KokkosKernels::Impl::Edge<int, Kokkos::complex<double>>>>>, _Size=long, _Compare=__gnu_cxx::__ops::_Iter_less_\
iter]"                                                                                                                                                                                                                  
/home/projects/ppc64le/gcc/6.4.0/include/c++/6.4.0/bits/stl_algo.h(1967): here                                                                                                                                          
            instantiation of "void std::__sort(_RandomAccessIterator, _RandomAccessIterator, _Compare) [with _RandomAccessIterator=__gnu_cxx::__normal_iterator<KokkosKernels::Impl::Edge<int, Kokkos::complex<double>>\
 *, std::vector<KokkosKernels::Impl::Edge<int, Kokkos::complex<double>>, std::allocator<KokkosKernels::Impl::Edge<int, Kokkos::complex<double>>>>>, _Compare=__gnu_cxx::__ops::_Iter_less_iter]"                        
/home/projects/ppc64le/gcc/6.4.0/include/c++/6.4.0/bits/stl_algo.h(4707): here                                                                                                                                          
            instantiation of "void std::sort(_RAIter, _RAIter) [with _RAIter=__gnu_cxx::__normal_iterator<KokkosKernels::Impl::Edge<int, Kokkos::complex<double>> *, std::vector<KokkosKernels::Impl::Edge<int, Kokkos:\
:complex<double>>, std::allocator<KokkosKernels::Impl::Edge<int, Kokkos::complex<double>>>>>]"                                                                                                                          
/home/kyukim/Work/lib/kokkoskernels/build/spotcheck/TestAll_2018-09-05_15.24.03/cuda/9.0.103/Cuda_OpenMP-release/install/include/KokkosKernels_SparseUtils.hpp(878): here                                               
            instantiation of "void KokkosKernels::Impl::kk_sort_graph<lno_view_t,lno_nnz_view_t,scalar_view_t,out_nnz_view_t,out_scalar_view_t,MyExecSpace>(lno_view_t, lno_nnz_view_t, scalar_view_t, out_nnz_view_t, \
out_scalar_view_t) [with lno_view_t=Kokkos::View<const int *, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::HostSpace::execution_space, Kokkos::HostSpace::memory_space>, Kokkos::MemoryTraits<0U>>, lno_nnz_view_t=Kokkos\
::View<int *, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::HostSpace::execution_space, Kokkos::HostSpace::memory_space>, Kokkos::MemoryTraits<0U>>, scalar_view_t=Kokkos::View<Kokkos::complex<double> *, Kokkos::LayoutR\
ight, Kokkos::Device<Kokkos::HostSpace::execution_space, Kokkos::HostSpace::memory_space>, void>, out_nnz_view_t=Kokkos::View<int *, Kokkos::LayoutLeft, Kokkos::Device<Kokkos::HostSpace::execution_space, Kokkos::Hos\
tSpace::memory_space>, Kokkos::MemoryTraits<0U>>, out_scalar_view_t=Kokkos::View<Kokkos::complex<double> *, Kokkos::OpenMP::array_layout, Kokkos::Device<Kokkos::HostSpace::execution_space, Kokkos::HostSpace::memory_\
space>, Kokkos::MemoryManaged>, MyExecSpace=Kokkos::OpenMP::execution_space]"                                                                                                                                           
/home/kyukim/Work/lib/kokkoskernels/master/unit_test/sparse/Test_Sparse_spgemm.hpp(195): here                                                                                                                           
            instantiation of "__nv_bool Test::is_same_matrix<crsMat_t,device>(crsMat_t, crsMat_t) [with crsMat_t=KokkosSparse::CrsMatrix<Kokkos::complex<double>, int, Kokkos::OpenMP, void, int>, device=Kokkos::OpenM\
P]"                                                                                                                                                                                                                     
/home/kyukim/Work/lib/kokkoskernels/master/unit_test/sparse/Test_Sparse_spgemm.hpp(374): here                                                                                                                           
            instantiation of "void test_spgemm<scalar_t,lno_t,size_type,device>(lno_t, size_type, lno_t, lno_t) [with scalar_t=Kokkos::complex<double>, lno_t=int, size_type=int, device=Kokkos::OpenMP]"               
/home/kyukim/Work/lib/kokkoskernels/master/unit_test/sparse/Test_Sparse_spgemm.hpp(445): here     
```

**Comment by srajama1** [comment]
Nope, looks new to me. Did something in Kokkos change or this is a new compiler ? SparseUtils hasn't changed.

**Comment by ndellingwood** [comment]
This looks related to PR kokkos/kokkos#1622 just merged today (09/05/2018), adding @ibaned

**Comment by crtrott** [comment]
This is something new. Someone requested swap, min and max as Kokkos functions so they work on GPUs ....

**Comment by crtrott** [comment]
Ok maybe we should just revert that and say that it is not a good idea. Or we need to rename the functions to something like `Kokkos::kokkos_swap()`. We did this for similar reasons with `malloc` and `free`.

**Comment by crtrott** [comment]
Nathan can you open an issue on the Kokkos side? If Dan is still there you can ask him what he thinks. I had some reservations about merging this for basically this reason: but today I thought I am being silly - "this is properly namespaced, its just gonna work ...."

**Comment by ndellingwood** [comment]
@ibaned took care of it with PR kokkos/kokkos#1781, kokkos develop branch should be clean once that is merged in.

**Comment by ndellingwood** [comment]
For reference the kokkos issue is kokkos/kokkos#1780

**Comment by ndellingwood** [comment]
Fixed by @ibaned with PR kokkos/kokkos#1781 and merged into develop branch.

**Comment by ndellingwood** [comment]
@srajama1 @crtrott can we add a label like 'bug-fixed-in-kokkos' to more accurately categorize the fix?

**Comment by srajama1** [comment]
We can, but this case is so transient I am fine just closing it.
